pages/events.js

```diff
@@ -40,18 +86,36 @@ export default function Events({ events }) {
       </Head>
       <Page>
         <h1 className="text-4xl mb-4 font-bold">Community events</h1>
-
-        <EventKey
-          categorisedEvents={categorisedEvents}
-          onToggleEventType={(newValue) => seteventType(newValue)}
+        <EventTabs
+          tabs={tabs}
+          setTabs={setTabs}
+          setSortEvents={setSortEvents}
+          sortEvents={sortEvents}
         />
 
-        {!events.length && <Alert type="info" message="No events found" />}
-        <ul role="list" className="divide-y divide-gray-200">
-          {categorisedEvents[eventType].map((event, index) => (
-            <EventCard event={event} username={event.username} key={index} />
-          ))}
-        </ul>
+        {filters.map(
+          (filter) =>
+            tabs.find((tab) => tab.key === filter.key) &&
```

Comment:
This could be simplifies if we have a `currentTab` state

Reply:
This is similar to how it's being used in the `[username].js`, I'll try to simplify it more
thanks for all these suggestions